### PR TITLE
complete constructors for function_application_exprt

### PR DESCRIPTION
**Pull request by kroening**
Fixing a problem uncovered by PR #3045.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by tautschnig — APPROVED** [review]

**Review by smowton — APPROVED** [review]

File: src/util/std_expr.h [hunk 1]
```diff
@@ -4333,22 +4333,16 @@ class null_pointer_exprt:public constant_exprt
...
-  }
+  using argumentst = exprt::operandst;
 
   function_application_exprt(
-    const symbol_exprt &_function, const typet &_type):
-      function_application_exprt(_type) // NOLINT(runtime/explicit)
+    const symbol_exprt &_function,
+    const argumentst &_arguments,
+    const typet &_type)
+    : binary_exprt(ID_function_application, _type) // NOLINT(runtime/explicit)
```
Comment (peterschrammel) [inline src/util/std_expr.h]: The NOLINT comment doesn't seem necessary anymore.

**Review by peterschrammel — APPROVED** [review]

File: src/util/std_expr.h (hunk 1 above)
Comment (tautschnig) [reply src/util/std_expr.h]: Removed.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: ec68b3b91c55c09b7ab6a53f58b570cc33f4cfbd).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85996359
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: ade7a78f3511bedadc2b95bbb34c57565d80887a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86003046

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: df1ee1c8cda77b5015ba5573103685a77aeac902).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86044590

**Pull request merged**